### DISCUSSION TITLE
Stuck at Filament plugin testing at resource edit page

### DISCUSSION THREAD

**Discussion by leekaimun** [filamentphp/filament#4986] [Help] [answered]
Encountered a problem while trying to test a Filament plugin that I am working on. To recreate the problem, I redo another plugin from scratch starting from filamentphp/plugin-skeleton. Then I put in the minimal plugin codes. The codes are available at https://github.com/leekaimun/testplugin.

Below is the test script and php threw an `ViewException Undefined variable $form ` while testing the `EditItem::class` component.

```
it('can test', function () {
    $name = 'Fake Name';
    livewire(CreateItem::class)
        ->fillForm(['name' => $name])
        ->call('create')
        ->assertHasNoFormErrors();  // Tested ok

    $this->assertDatabaseHas(Item::class, ['name' => $name]); // Tested ok

    livewire(ListItem::class)->assertSee($name); // Tested ok

    $record = Item::where('name', $name)->first();
    livewire(EditItem::class, ['record' => $record->id]);
    /*
      The above code encountered Illuminate\View\ViewException
      Undefined variable $form
      (View: .../testplugin/vendor/filament/filament/resources/views/resources/pages/edit-record.blade.php)
     */
});
```

When I created a new laravel project and installed this plugin, everything worked fine.

Wonder if anyone encountered such a problem?

Laravel: v9.40.1
Filament: v2.16.50
Orchestra: v7.13.0

**Comment by danharrin** [comment, marked as answer]
Heres a list of providers that need importing: https://github.com/filamentphp/filament/blob/2.x/tests/src/TestCase.php#L27-L37

**Comment by leekaimun** [comment]
Thanks, as always!
